data-access-layer/service-agent/Agent.js

```diff
@@ -200,6 +200,29 @@ class Agent extends HttpClient {
       .then(ip => this.post(ip, 'lifecycle/preupdate', context, CONST.HTTP_STATUS_CODE.OK, agentCredsBeforeUpdate));
   }
 
+  /**
+   * Poll the agent for the process of the lifecycle last operation within the deployment
+   *
+   * @param ips The ip addresses of all available agents
+   * @param {string} eventType The current lifecycle event, supported values: "create", "update"
+   * @param {string} lifecycleState The current lifecycle state of the operation, supported values: "post"
+   * @returns {*} The response of the agent
+   */
+  getProcessingState(ips, eventType, lifecycleState) {
```

Comment:
Can we name the feature as lifecycle.post[create|update] ? We already have `lifecycle.preupdate`. In that case, it will be consistent. You can then handle this in a generic way.
`lifecycleState` can then support pre and post both.